validation.lua

```diff
@@ -118,5 +124,12 @@ create_token = function (self, purpose, username, email)
         email,
         mail_subjects[purpose] .. username,
         mail_bodies[purpose],
-        self:build_url('/users/' .. username .. '/' .. purpose .. '/' .. token_value))
+        self:build_url(self:url_for(
+            purpose,
+            {
+                username = url.escape(username),
```

Comment:
This works well, though it seems to encode characters that are "URL safe", like `-`. Do you know if there's a way around that? If not, this is still a clear improvement, but just trying to consider some future uses.

Reply:
Yeah, good call. [`url.build_path()`](http://w3.impa.br/~diego/software/luasocket/url.html#build_path) is what we want.